ansible_collections/arista/avd/plugins/plugin_utils/topology_generator_utils.py

```diff
@@ -0,0 +1,314 @@
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+import os
+import yaml
+import queue
+import graphviz
+
+def read_yaml_file(filename):
+    """
+    Open a structure config file and load a data from the file
+
+    Args:
+        filename: Name of the file
+
+    Returns:
+        Data from the file
+    """
+    with open(filename, 'r', encoding='utf8') as stream:
+        data = None
+        try:
+            data = yaml.safe_load(stream)
+        except yaml.YAMLError as exc:
+            print(exc)
+        return data
+
+
+
+def create_node_dict(file_data, filename):
+    """
+    Create a dictionary of nodes with its corresponding neighbours
+
+    Args:
+        file_data: Data of structure config file
+        filename: Name of the file
+
+    Returns:
+        Dictionary with node and its corresponding neighbours
+    """
+    node_dict = {}
+    node_dict['name'] = os.path.splitext(filename)[0].split('/')[2]
+    node_dict['neighbours'] = []
+    if file_data and 'ethernet_interfaces' in list(file_data.keys()) and file_data['ethernet_interfaces']:
+        for key, value in file_data['ethernet_interfaces'].items():
+            neighbour_dict = {}
+            neighbour_dict['neighborDevice'] = value['peer']
+            neighbour_dict['neighborPort'] = value['peer_interface']
+            neighbour_dict['port'] = key
+            if 'channel_group' in value and 'id' in value['channel_group']:
+                neighbour_dict['portChannel'] = value['channel_group']['id']
+            node_dict['neighbours'].append(neighbour_dict)
+    return node_dict
+
+
+def structured_config_to_topology_input(output_list, node_dict, diagram_groups, current_dict=None):
+    """
+    Convert structure config file data to topology input data by creating a nested dictionary
+    of "nodes" and "groups"
+
+    Args:
+        output_list: Original input dictionary with nested "name", "nodes" and "groups" keys
+        nodes: List of nodes
+        node_neighbour_dict: Dictionary with nodes and neighbours list
+                       ex- {'0': ['super-spine1', 'super-spine2']}
+    Returns:
+        nodes: List of nodes which are present in the graph
+        node_neighbour_dict: Dictionary of node and its corresponding neighbours
```

Comment:
I checked the returned values ,looks like the same. Please let me know if I am missing anything.
I checked the values of the nodes... ['0', 'super-spine1', 'super-spine2', 'spine2', 'l3-leaf1', 'spine1', 'l3-leaf2', 'l2-leaf1', 'spine3', 'l3-leaf3', 'spine4', 'spine5', 'l3-leaf5'.......] and 

node_neighbour_dict... {'0': ['super-spine1', 'super-spine2', 'super-spine3', 'super-spine4'], ....]

Reply:
The last statement is `return output_list` so the function only returns a list.

It also updates the node_neighbour_dict but it is not a returned value which has a different meaning in docstring.

If you had to type the function it would be:

```
def structured_config_to_topology_input(output_list: list, node_dict: dict, diagram_groups, current_dict=None) -> list:
```

Also it looks like output_list is also one of the input parameters so the idea is to take as input the list and modify it (or create if it was None or empty) and then output it back?

Reply:
My bad! I have checked again and made changes in the docstring.  Yes, Idea is to take as input the list and modify it (or create if it was None or empty) and then output it back. We are updating the same in a recursive manner.